includes/class-stats.php

```diff
@@ -3149,9 +3023,89 @@ private function set_date_ranges() {
 					break;
 			}
 
-			$dates['range'] = $range;
+			if ( ! empty( $dates ) ) {
+				// Convert the values to the UTC equivalent so that we can query the database using UTC.
+				$dates['start'] = edd_get_utc_equivalent_date( $dates['start'] );
+				$dates['end']   = edd_get_utc_equivalent_date( $dates['end'] );
+				$dates['range'] = $range;
+
+				$this->relative_date_ranges[ $range ] = $dates;
+			}
+		}
+
+	}
+
+	/**
+	 * Based on the query_vars['revenue_type'], use gross or net statuses.
+	 *
+	 * @since 3.0
+	 *
+	 * @return array The statuses of orders to use for the stats generation.
+	 */
+	private function get_revenue_type_statuses() {
+		if ( 'net' === $this->query_vars['revenue_type'] ) {
+			return edd_get_net_order_statuses();
+		}
+
+		return edd_get_gross_order_statuses();
+	}
+
+	/**
+	 * Based on the query_vars['revenue_type'], use just sale or also include refunds.
+	 *
+	 * @since 3.0
+	 *
+	 * @return array The order types to use when generating stats.
+	 */
+	private function get_revenue_type_order_types() {
+		$order_types = array( 'sale' );
+		if ( 'net' === $this->query_vars['revenue_type'] ) {
+			$order_types[] = 'refund';
+		}
+
+		return $order_types;
+	}
+
+	/**
+	 * Generates output for the report tiles when a relative % change is requested.
+	 *
+	 * @since 3.0
+	 *
+	 * @param int|float $total     The primary value result for the stat.
+	 * @param int|float $relative  The value relative to the previous date range.
+	 * @param bool      $reverse   If the stat being displayed is a 'reverse' state, where lower is better.
+	 */
+	private function generate_relative_markup( $total = 0, $relative = 0, $reverse = false ) {
+		$relative_markup  = '';
+
+		$total_output    = $this->maybe_format( $total );
+		$relative_output = '';
+
+		if ( ( floatval( 0 ) === floatval( $total ) && floatval( 0 ) === floatval( $relative ) ) || ( $total === $relative ) ) {
+			$relative_output = esc_html__( 'No Change', 'easy-digital-downloads' );
```

Comment:
```suggestion
			return $relative_markup;
```
Suggesting this as I think returning an empty string here might reduce confusion as to why sometimes a tile will show a trend/percentage change, sometimes "No Change", and sometimes nothing at all. If it only shows a trend or nothing at all, that seems potentially less confusing.

Reply:
These are three distinct cases though, and I think `No Change` is actually important to show.

* Trend means we had data to compare against and there was a difference
* No Change means we had data to compare against and there was no difference
* Showing nothing at all means we didn't have any data to compare against.

Removing 'No Change' is factually incorrect. If we are going to change something I think it should be what we show when we have no data to compare against. We could just use an mdash or something similar to show that we don't have an entry for this.